scripts/globals/automatonweaponskills.lua

```diff
@@ -150,27 +156,31 @@ function doAutoPhysicalWeaponskill(attacker, target, wsID, tp, primary, action,
         local chance = math.random();
         if ((chance<=hitrate or math.random() < attacker:getMod(MOD_ZANSHIN)/100 or isSneakValid)
                 and not target:hasStatusEffect(EFFECT_PERFECT_DODGE) and not target:hasStatusEffect(EFFECT_ALL_MISS) ) then -- it hit
-            pdif = generateAutoPdif(cratio[1], cratio[2], true);
-            if (params.canCrit) then
-                critchance = math.random();
-                if (critchance <= nativecrit or hasMightyStrikes) then -- crit hit!
-                    criticalHit = true;
-                    cpdif = generateAutoPdif(ccritratio[1], ccritratio[2], true);
-                    finaldmg = finaldmg + dmg * cpdif;
+            if not shadowAbsorbed(target) then
```

Comment:
this whole block (the offhand damage block) can never be true, might as well get rid of it all